geode-core/src/main/java/org/apache/geode/internal/cache/FilterProfile.java

```diff
@@ -1846,18 +1846,27 @@ protected void process(ClusterDistributionManager dm) {
         }
 
         CacheDistributionAdvisor cda = (CacheDistributionAdvisor) r.getDistributionAdvisor();
-        CacheDistributionAdvisor.CacheProfile cp =
-            (CacheDistributionAdvisor.CacheProfile) cda.getProfile(getSender());
-        if (cp == null) { // PR accessors do not keep filter profiles around
-          if (logger.isDebugEnabled()) {
-            logger.debug(
-                "No cache profile to update, adding filter profile message to queue. Message :{}",
-                this);
+        CacheDistributionAdvisor.CacheProfile cp;
+
+        // prevent adding the message to queue after we have processed the queue
+        // in CreateRegionReplyProcessor.process
+        synchronized (cda) {
```

Comment:
Jinmei, what happens if the CreateRegionMessageReply is processed before the lock is acquired. Is it possible, does the cache profile in reply has the information about the registered CQs in server1?

Reply:
I can see that CacheProfile has the FilterProfile.

Reply:
If the CreateRegionMessageReply is processed before the lock is acquired (more precisely, if the putProfile call is processed inside the `process` method), then the if(pf==null) will return false, so the register_cq message will not be put in the queue, but to be processed regularly.